client/python/openlineage/client/transport/http.py

```diff
@@ -93,7 +93,8 @@ def __init__(self, config: HttpConfig):
             raise ValueError(f"Need valid url for OpenLineageClient, passed {url}. Exception: {e}")
         self.url = url
         self.session = config.session
-        self.session.headers['Content-Type'] = 'application/json'
+        if self.session:
```

Comment:
I think the bug here is actually the fact that session is optional in `HttpConfig`. It's given a default factory, so it will never be None.

The type for `session` should be changed to be without `Optional`.